ChromiumHtmlToPdfLib/Connection.cs

```diff
@@ -119,27 +114,30 @@ internal Connection(string url, int timeout, Logger logger)
         _webSocket = new ClientWebSocket();
         _receiveLoopCts = new CancellationTokenSource();
         OpenWebSocketAsync().GetAwaiter().GetResult();
-        Task.Factory.StartNew(ReceiveLoop, _receiveLoopCts.Token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
+        Task.Factory.StartNew(ReceiveLoop, new ReceiveLoopState(_logger, _webSocket, OnMessageReceived, _receiveLoopCts.Token), _receiveLoopCts.Token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
     }
     #endregion
 
     #region ReceiveLoop
-    private async Task ReceiveLoop()
+    private sealed record ReceiveLoopState(Logger? Logger, ClientWebSocket WebSocket, Action<string> OnMessageReceived, CancellationToken Token);
+
+    private static async Task ReceiveLoop(object? stateData)
     {
-        var loopToken = _receiveLoopCts.Token;
-        MemoryStream outputStream = null;
+        var state = (ReceiveLoopState)stateData!;
+
+        MemoryStream? outputStream = null;
         var buffer = new ArraySegment<byte>(new byte[ReceiveBufferSize]);
 
         try
         {
-            while (!loopToken.IsCancellationRequested)
+            while (!state.Token.IsCancellationRequested)
             {
                 outputStream = new MemoryStream(ReceiveBufferSize);
                 WebSocketReceiveResult receiveResult;
 
                 do
                 {
-                    receiveResult = await _webSocket.ReceiveAsync(buffer, _receiveLoopCts.Token).ConfigureAwait(false);
```

Comment:
was: potantial race condition on _receiveLoopCts field (should have used loopToken)